<launch>
    <arg name="veh"/>
    <arg name="pkg_name" value="pid_controller"/>
    <arg name="node_name" default="pid_controller_node"/>
    <arg name="param_file_name" default="default" doc="Specify a param file. ex:default." />

    <group ns="$(arg veh)">
        <remap from="$(arg node_name)/mode" to="flight_controller_node/mode/current" />
        <remap from="$(arg node_name)/commands" to="mavros/setpoint_raw/attitude" />
        <remap from="$(arg node_name)/state" to="state_estimator_node/state" />
        <remap from="$(arg node_name)/set_mode" to="flight_controller_node/set_mode" />

        <node name="$(arg node_name)" pkg="$(arg pkg_name)" type="$(arg node_name).py" output="screen">
            <rosparam command="load" file="$(find pid_controller)/config/$(arg node_name)/$(arg param_file_name).yaml"/>
        </node>
    </group>
</launch>

